prism-agent/service/server/src/main/scala/io/iohk/atala/issue/controller/IssueControllerImpl.scala

```diff
@@ -115,6 +128,40 @@ class IssueControllerImpl(
     mapIssueErrors(result)
   }
 
+  private def validatePrismDID(
+      prismDID: PrismDID,
+      allowUnpublished: Boolean
+  ): ZIO[WalletAccessContext, ErrorResponse, Unit] = {
+    val result = for {
+      maybeDIDState <- managedDIDService.getManagedDIDState(prismDID.asCanonical)
+      maybeMetadata <- didService.resolveDID(prismDID).map(_.map(_._1))
+      result <- (maybeDIDState.map(_.publicationState), maybeMetadata.map(_.deactivated)) match {
```

Comment:
I would refactor this code to use a single-dimensional pattern-matching. As for me, it's hard to read and track all the cases.

Reply:
Yes, I first tried to implement it as you suggested, but it was even more complex to grasp. I found it clearer to match the two relevant states (publication and activation) side by side, given they are interdependent here. So I propose to keep it like that for now.